components/ac_log/android/src/main/java/mozilla/appservices/rustlog/LibRustLogAdapter.kt

```diff
@@ -0,0 +1,62 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.appservices.rustlog
+
+import android.util.Log
+import com.sun.jna.Callback
+import com.sun.jna.Library
+import com.sun.jna.Native
+import com.sun.jna.Pointer
+import com.sun.jna.PointerType
+import java.lang.reflect.Proxy
+
+@Suppress("FunctionNaming", "TooManyFunctions", "TooGenericExceptionThrown")
+internal interface LibRustLogAdapter : Library {
+    companion object {
+        private val JNA_LIBRARY_NAME = {
+            val libname = System.getProperty("mozilla.appservices.ac_rust_log_lib_name")
+            if (libname != null) {
+                Log.i("AppServices", "Using ac_rust_log_lib_name: " + libname);
+                libname
+            } else {
+                "ac_rust_log"
+            }
+        }()
+
+        internal var INSTANCE: LibRustLogAdapter = try {
+            Native.loadLibrary(JNA_LIBRARY_NAME, LibRustLogAdapter::class.java) as LibRustLogAdapter
+        } catch (e: UnsatisfiedLinkError) {
```

Comment:
It was just cargo culted from fxa-client. In particular though, if nobody enables logging, I don't want things to break just because the log library can't be linked.

Reply:
I've fixed this to not use it, and also to use direct dispatch which should be a lot faster.